infra/src/main/kotlin/fi/fta/geoviite/infra/geometry/GeometryValidation.kt

```diff
@@ -289,10 +289,16 @@ fun validateKmPosts(kmPosts: List<GeometryKmPost>): List<GeometryValidationIssue
 private fun validateKmPostCollection(kmPosts: List<GeometryKmPost>): List<CollectionIssue> {
     val groupedKmPosts = kmPosts.filter { it.kmNumber != null }.sortedBy { it.kmNumber }
     val firstKmPost = groupedKmPosts.firstOrNull()
+    val duplicateKmPosts = groupedKmPosts.groupBy { it.kmNumber }.filter { it.value.size > 1 }.keys
 
     val generalErrors = listOfNotNull(
-        validate(groupedKmPosts.filter { kmPost -> kmPost.kmNumber == firstKmPost?.kmNumber }.size == 1) {
-            CollectionIssue("multiple-start-km-posts", VALIDATION_KM_POST, VALIDATION_ERROR, firstKmPost?.kmNumber?.toString())
+        validate(duplicateKmPosts.isEmpty()) {
+            CollectionIssue(
+                "duplicate-km-posts",
+                VALIDATION_KM_POST,
+                VALIDATION_ERROR,
+                duplicateKmPosts.map { it.toString() }.joinToString(", ")
+            )
```

Comment:
Aiemmin siis on jo tehty vähän tämäntyyppinen tarkastelu, mutta se on tehty ainoastaan alignmentin aloittavalle kilometripylväälle. Tuo vanha validaatio on nyt yhdistetty tähän uuteen, sillä ensimmäisessä kilometripylväässä ei pitäis olla tältä kannalta mitään spesiaalia.